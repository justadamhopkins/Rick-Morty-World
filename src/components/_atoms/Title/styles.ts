import { motion } from "framer-motion";
import styled from "styled-components";

export const StyledTitle = styled(motion.h1)`
  text-align: center;
  background-size: 100% 200%;
  background-image: linear-gradient(
    var(--secondaryBackground) 50%,
    var(--background) 50%
  );
  padding-top: 20px;
  display: inline-block;
  transform: skew(-5deg);
  color: var(--darkPurps);
  padding: 5px;
  margin: 0px 15px;
`;
