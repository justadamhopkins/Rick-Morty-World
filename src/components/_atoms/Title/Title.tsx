import { FC } from "react";
import { StyledTitle } from "./styles";

interface Props {
  delay?: number;
  copy: string;
}

export const Title: FC<Props> = ({ copy, delay = 1 }) => {
  return (
    <StyledTitle
      initial={{ backgroundPosition: "0 100%" }}
      animate={{ backgroundPosition: "0% 0%" }}
      transition={{
        duration: 0.7,
        ease: "easeIn",
        delay,
      }}
    >
      {copy}
    </StyledTitle>
  );
};
