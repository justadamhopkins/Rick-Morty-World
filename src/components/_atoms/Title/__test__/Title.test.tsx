import { render } from "@testing-library/react";
import { Title } from "../Title";

describe("Title", () => {
  const renderTitle = () => {
    return render(<Title copy="mockCopy" />);
  };
  test("should render a single card component", () => {
    const { asFragment } = renderTitle();

    expect(asFragment()).toMatchSnapshot();
  });
});
