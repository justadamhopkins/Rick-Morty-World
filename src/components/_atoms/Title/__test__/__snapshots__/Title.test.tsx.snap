// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title should render a single card component 1`] = `
<DocumentFragment>
  <h1
    class="styles__StyledTitle-b4jvka-0 gVORke"
  >
    mockCopy
  </h1>
</DocumentFragment>
`;
